src/equisolve/numpy/models/linear_model.py

```diff
@@ -32,6 +36,20 @@ class Ridge:
     :param parameter_keys:
         Parameters to perform the regression for. Examples are ``"values"``,
         ``"positions"``, ``"cell"`` or a combination of these.
+
+    :param solver:
+        Parameters that determines which solver is used for solving the linear system to
+        obtain the weights. We support
+        "cholesky": using scipy.linalg.solve on (X.T@X) w = X.T @ y
```

Comment:
```suggestion
        "cholesky": using :func:`scipy.linalg.solve` on (X.T@X) w = X.T @ y
```

Reply:
Why not use https://numpy.org/doc/stable/reference/generated/numpy.linalg.cholesky.html instead of `scipy.linalg.solve`? Also, this would use `torch.linalg.solve` with torch I guess?

Reply:
We also allow different alpha values for individual feature (scikit-learn does not do this), that makes the logic in the svd path a bit different, so we cannot just use scikits cholesky function. On the other hand, we could drop this individual alpha value option and add a feature scaler that does this.

But more importantly, if we want to replace this with equistore operations at some point, I think we need to have the logic inside our function, so we can use the equistore solver function later that dispatches to the corresponding solver.

I also like that we do not have a scikit-learn dependency.



Yes so integrating torch here would be a